Assets/BossRoom/Scripts/Shared/ApplicationController.cs

```diff
@@ -88,7 +88,15 @@ private void OnDestroy()
         /// </summary>
         private IEnumerator LeaveBeforeQuit()
         {
-            m_LobbyServiceFacade.ForceLeaveLobbyAttempt();
+            // We want to quit anyways, so if anything happens while trying to leave the Lobby, log the exception then carry on
+            try
+            {
+                m_LobbyServiceFacade.ForceLeaveLobbyAttempt();
```

Comment:
Doesn't this method just end up calling `LobbyServiceFacade.EndTracking`? 

So I was looking into this this [bug](https://jira.unity3d.com/browse/MTT-2799) here, and I found success when hooking into the `onFailure` Action on some of these Lobby API calls, even when in an offline state. And looking at `EndTracking()`, it could only possibly return an `AsyncException` when trying to leave the lobby and failing. I have a suspicion that is the error we're seeing here. Wouldn't it make sense to just call `EndTracking()` here, modify `EndTracking` to allow passing in a `onFailure` action, and here quit the application when failing? @SamuelBellomo @pdeschain

Reply:
This is likely another bug. From the screenshots we had regarding this bug, the users had `NullReferenceException`s. If returning an AsyncException when failing to leave the lobby, wouldn't that not impact the flow here since it is done asynchronously?

Reply:
I see both on the referenced Jira ticket. It's a shame we didn't capture the `NullReferenceException`s.

Since it's hard to reproduce, why not include a try-catch inside of `OnWantToQuit()`. As well, why are we starting the `LeaveBeforeQuit()` coroutine even when `canQuit` is `true`?

Reply:
And my first point is that there's a redundancy in the two methods. `ForceLeaveLobbyAttempt()` just calls `EndTracking()`. Why not call `EndTracking()` directly?